tools/clang/tools/dxc/dxc.cpp

```diff
@@ -1017,10 +1017,22 @@ void DxcContext::GetCompilerVersionInfo(llvm::raw_string_ostream &OS) {
     UINT32 compilerMajor = 1;
     UINT32 compilerMinor = 0;
     CComPtr<IDxcVersionInfo> VerInfo;
+
+#ifdef SUPPORT_QUERY_GIT_COMMIT_INFO
+    UINT32 commitCount = 0;
+    CComHeapPtr<char> commitHash;
+    CComPtr<IDxcVersionInfo2> VerInfo2;
+#endif // SUPPORT_QUERY_GIT_COMMIT_INFO
+
     const char *compilerName =
         m_Opts.ExternalFn.empty() ? "dxcompiler.dll" : m_Opts.ExternalFn.data();
+
     if (SUCCEEDED(CreateInstance(CLSID_DxcCompiler, &VerInfo))) {
       VerInfo->GetVersion(&compilerMajor, &compilerMinor);
+#ifdef SUPPORT_QUERY_GIT_COMMIT_INFO
+      if (SUCCEEDED(CreateInstance(CLSID_DxcCompiler, &VerInfo2)))
```

Comment:
Doesn't matter very much, but you need not create another compiler object - you can simply QueryInterface on VerInfo to get VerInfo2.

Reply:
Ah, that's good suggestion. Done.